rusk-wallet/src/clients.rs

```diff
@@ -53,188 +36,87 @@ const STAKE_CONTRACT: &str =
-        address: BlsScalar,
-        signature: Signature,
-    ) -> Result<Proof, Self::Error> {
-        let mut buf = [0; STCT_INPUT_SIZE];
-        let mut writer = &mut buf[..];
-        writer.write(&fee.to_bytes())?;
-        writer.write(&crossover.to_bytes())?;
-        writer.write(&value.to_bytes())?;
-        writer.write(&blinder.to_bytes())?;
-        writer.write(&address.to_bytes())?;
-        writer.write(&signature.to_bytes())?;
-
-        self.status("Requesting stct proof...");
-
-        let prove_req = RuskRequest::new("prove_stct", buf.to_vec());
-        let res = self.prover.call(2, "rusk", &prove_req).wait()?;
-
-        self.status("Stct proof success!");
-
-        let mut proof_bytes = [0u8; Proof::SIZE];
-        proof_bytes.copy_from_slice(&res);
-
-        let proof = Proof::from_bytes(&proof_bytes)?;
-        Ok(proof)
-    }
+/// A prover struct that has the `Prove` trait from executio-core implemented.
+/// It currently uses a hardcoded prover which delegates the proving to the
+/// `prove_execute`
+pub struct Prover;
 
-    /// Request a WFCT proof.
-    fn request_wfct_proof(
-        &self,
-        commitment: JubJubAffine,
-        value: u64,
-        blinder: JubJubScalar,
-    ) -> Result<Proof, Self::Error> {
-        let mut buf = [0; WFCT_INPUT_SIZE];
-        let mut writer = &mut buf[..];
-        writer.write(&commitment.to_bytes())?;
-        writer.write(&value.to_bytes())?;
-        writer.write(&blinder.to_bytes())?;
-
-        self.status("Requesting wfct proof...");
-        let prove_req = RuskRequest::new("prove_wfct", buf.to_vec());
-        let res = self.prover.call(2, "rusk", &prove_req).wait()?;
-        self.status("Wfct proof success!");
-
-        let mut proof_bytes = [0u8; Proof::SIZE];
-        proof_bytes.copy_from_slice(&res);
-
-        let proof = Proof::from_bytes(&proof_bytes)?;
-        Ok(proof)
-    }
-}
+impl Prove for Prover {
+    fn prove(tx_circuit_vec_bytes: &[u8]) -> Result<Vec<u8>, WalletCoreError> {
+        let prover = RuskHttpClient::new("http://127.0.0.1:8080".to_string());
+        let prove_req =
+            RuskRequest::new("prove_execute", tx_circuit_vec_bytes.to_owned());
 
-impl Prover {
-    fn status(&self, text: &str) {
-        (self.status)(text)
+        prover
+            .call(2, "rusk", &prove_req)
+            .wait()
+            .map_err(|e| WalletCoreError::PhoenixCircuit(e.to_string()))
     }
 }
 
-/// Implementation of the StateClient trait from wallet-core
-/// inner is an option because we don't want to open the db twice and lock it
-/// We construct StateStore twice
-pub struct StateStore {
-    inner: Mutex<InnerState>,
+/// The state struct is responsible for managing the state of the wallet
+pub struct State {
+    cache: Mutex<Arc<Cache>>,
     status: fn(&str),
-    pub(crate) store: LocalStore,
-}
-
-struct InnerState {
     client: RuskHttpClient,
-    cache: Arc<Cache>,
+    addresses: Addresses,
```

Comment:
To me the keys are not part of the `State` but really should be sandboxed in the wallet. With this we should also decide whether we want to store the seed and generate the keys each time we need them *or* if we generate all keys in the beginning and zeroize the seed. But I would like to avoid keeping both around in two different places.

Reply:
I bought back the local store and we pass the reference around